tower-route/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+extern crate futures;
+extern crate tower_layer;
+extern crate tower_service;
+extern crate void;
+
+use futures::{Async, Poll};
+use tower_service::Service;
+
+pub trait Acceptor<Request> {
+    fn accepts(&self, request: &Request) -> bool;
+}
+
+pub struct Route<'a, A, B> {
+    service: &'a mut A,
+    other: &'a mut B,
+}
+
+impl<'a, A, B, Request, Response, Future, Error> Service<Request> for Route<'a, A, B>
+where
+    Future: futures::Future<Item = Response, Error = Error>,
+    A: Service<Request, Response = Response, Future = Future, Error = Error> + Acceptor<Request>,
```

Comment:
might make more sense to have a separate acceptor struct, instead of forcing this to be bundled with `Service`? This wouldn't say allow you to erase the type into a `impl Service` I believe.

Reply:
Sorry for the delay. I've made the other suggestions but I'm not exactly sure what you mean by this one.